docs/source/index.rst

```diff
@@ -2,54 +2,101 @@
-  processing (`scikit-image`_) , and genomics (`scikit-allel`_).
-* Dask enables data scientists to process tens of gigabytes of data on a laptop
-  using familiar NumPy and Pandas interfaces.
-* Dask enables researchers to rapidly prototype and deploy novel algorithms on a
-  single machine or a distributed cluster.
 
-Dask user intro
-===============
+Familiar user interface
+-----------------------
 
-Users interact with dask either by making graphs directly, or through the
-*dask collections* which provide larger-than-memory counterparts to existing
-popular libraries:
+**Dask DataFrames** mimic Pandas
 
-* ``dask.array`` = ``numpy`` + ``threading``
-* ``dask.bag`` = ``map, filter, toolz`` + ``multiprocessing``
-* ``dask.dataframe`` = ``pandas`` + ``threading``
+.. code-block:: python
 
-Dask primarily targets parallel computations that run on a single machine. It
-integrates nicely with the existing PyData ecosystem and is trivial to setup
-and use::
+    import pandas as pd                     import dask.dataframe as dd
+    df = pd.read_csv('2015-01-01.csv')      df = dd.read_csv('2015-*-*.csv')
+    df.groupby(df.user_id).value.mean()     df.groupby(df.user_id).value.mean().compute()
 
-    conda install dask
-    or
-    pip install dask
+**Dask Arrays** mimic NumPy
 
-Operations on dask collections (array, bag, dataframe) produce task graphs that
-encode blocked algorithms. Task schedulers execute these task graphs in
-parallel in a variety of contexts.
+.. code-block:: python
 
-.. image:: images/collections-schedulers.png
-   :alt: Dask collections and schedulers
-   :width: 80%
-   :align: center
+   import numpy as np                       import dask.array as da
+   f = h5py.File('myfile.hdf5')
+   x = np.array(f['/small-data'])           x = da.from_array(f['/big-data'],
+                                                              chunks=(1000, 1000))
+   x - x.mean(axis=1)                       x - x.mean(axis=1).compute()
+
+**Dask Bags** mimic iterators, Toolz, PySpark
+
+.. code-block:: python
+
+   import dask.bag as db
+   b = db.from_filenames('2015-*-*.json.gz').map(json.loads)
+   b.pluck('name').frequencies().topk(10, lambda pair: pair[1]).compute()
+
+**Dask Imperative** mimic for loops
+
+.. code-block:: python
+
+   from dask.imperative import do
+   L = []
+   for fn in filenames:            # Use for loops to build up computation
+       data = do(load)(fn)         # Delay execution of function with `do`
+       L.append(do(process)(data)) # Build connections between variables
+
+   result = do(summarize)(L)
+   result.compute()
+
+
+Scales from laptops to clusters
+-------------------------------
+
+Dask is convenient on a laptop.  It :doc:`installs <install>` trivially with
+``conda`` or ``pip`` and extends the size of convenient datasets from "fits in
+memory" to "fits on disk".
+
+Dask is powerful to run on a cluster of 100s of machines. It is resilient,
```

Comment:
"Dask is powerful to run on a cluster..." -> "dask is powerful enough to run on a cluster..."?